src/utils/signTypedData.ts

```diff
@@ -15,11 +15,21 @@ JsonRpcSigner.prototype._signTypedData = async function signTypedDataWithFallbac
   const address = await this.getAddress()
 
   try {
-    return await this.provider.send('eth_signTypedData_v4', [
-      address.toLowerCase(),
-      JSON.stringify(_TypedDataEncoder.getPayload(populated.domain, types, populated.value)),
-    ])
+    try {
+      return await this.provider.send('eth_signTypedData', [
+        address.toLowerCase(),
+        JSON.stringify(_TypedDataEncoder.getPayload(populated.domain, types, populated.value)),
+      ])
+    } catch (error) {
+      if (typeof error.code === 'number' && error.code === -32602) {
```

Comment:
can you leave a comment describing the error code above this line? the `typeof` check also feels unnecessary

Reply:
typeof check should be necessary because error is unknown.

Reply:
i dont get it, `error.code === -32602` implicitly implies it's a number already

Reply:
if `error` is unknown then you should be type guarding on the `error` object, not the internal `.code` property. for example if `error` is a primitive like `1` then this would fail - not a huge deal since we expect error formats to be relatively consistent, but at runtime this check isn't really doing anything